thorlcr/activities/piperead/thprslave.cpp

```diff
@@ -522,6 +520,14 @@ PipeWriterThread::PipeWriterThread(CPipeThroughSlaveActivity & _activity)
     inputStream = activity.inputStream;
 }
 
+IException *PipeWriterThread::joinExc()
+{
+    Thread::join();
+    if (activity.verifyPipeException) // if verifyPipe failure, ignore [follow-on] pipe write failure.
```

Comment:
Will verify pipe have been called before this function?  It looks to me that it is called after (in CPipeThroughSlaveActivity::stop).
Possibly a cleaner fix is

        if (wrexc && !verifyPipeException)

in CPipeThroughSlaveActivity::stop, since the writer shouldn't really need to access activity.

Reply:
verifyPipe will [normally] have been called before this on error, via CPipeStream::read() (via nextRow).
The call to verifyPipe in stop() will normally be a nop (pipeFinished already true).
But if nextRow is never called, then verifyPipe is called afterwards, but then the exception it generates would be thrown directly (before wrexc).

I agree though that testing verifyPipeException in stop() would be better.